plugins/README.md

```diff
@@ -0,0 +1,246 @@
+
+However, plugins should be standalone and work independently from one another, with or without the Gutenberg plugin enabled.
+
+Currently, the main use for canonical plugins in this repository is for single block plugins.
+
+Note that `PLUGIN_NAME` and `PLUGIN-SLUG` are used here as placeholders and should be replaced with the actual plugin name and slug in the following example code.
+
+## Folder structure
+
+`plugins/PLUGIN-SLUG/` directory:
+```
+- .npmrc
+- assets/
+	- screenshot-*.png
+- block.json (for block plugins)
+- package.json
+- phpcs.xml.dist
+- readme.txt
+- src/
+	- index.js
+	- style.scss
+- PLUGIN-SLUG.php
+```
+
+### Plugin Header
+
+The header in the plugin's main .php file should use the [standard plugin headers](https://developer.wordpress.org/plugins/plugin-basics/header-requirements/), with the author as "WordPress.org".
+
+```php
+<?php
+/**
+ * Plugin Name:       Plugin Name
+ * Plugin URI:        https://github.com/WordPress/gutenberg/tree/trunk/plugins/PLUGIN-SLUG
+ * Description:       A description of the plugin.
+ * Version:           1.0.0
+ * Requires at least: 6.4
+ * Requires PHP:      7.0
+ * Author:            WordPress.org
+ * Author URI:        https://wordpress.org/
+ * License:           GPL-2.0-or-later
+ * License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+ * Text Domain:       PLUGIN-SLUG
+ *
+ */
+```
+
+### package.json
+
+- Plugins aren't published to npm, so should be marked as private in the package.json settings.
+- `@wordpress/*` packages should specify a `file:` reference to the relevant package in the monorepo.
+- Only declare scripts that are unique to the plugin, such as `"build"` and `"plugin-zip"`. Scripts for linting, formatting, tests, etc are shared across the repo.
+
+```json
+{
+	"private": true,
+	"name": "PLUGIN-SLUG",
+	"version": "1.0.0",
+	"description": "A description of the plugin",
+	"author": "WordPress.org",
+	"license": "GPL-2.0-or-later",
+	"keywords": [ "wordpress", ... ],
+	"homepage": "https://github.com/WordPress/gutenberg/tree/trunk/plugins/PLUGIN-SLUG",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/WordPress/gutenberg.git",
+		"directory": "plugins/PLUGIN-SLUG"
+	},
+	"bugs": {
+		"url": "https://github.com/WordPress/gutenberg/issues"
+	},
+	"main": "build/index.js",
+	"dependencies": {
+		"@wordpress/block-editor": "file:../../packages/block-editor",
+		...
+	},
+	"publishConfig": {
+		"access": "public"
+	},
+	"scripts": {
+		"build": "wp-scripts build",
+		"plugin-zip": "wp-scripts plugin-zip",
+		"start": "wp-scripts start"
+	}
+}
+```
+
+## Block Plugins
+
+Block plugins should follow the [block specifec plugin guidelines](https://developer.wordpress.org/plugins/wordpress-org/block-specific-plugin-guidelines/#block-plugins-and-the-block-directory) and contain only one top-level block per plugin.
```

Comment:
I can see cases where multiple blocks can be in the same plugin: example a form block with its children blocks. How should these be handled. (Doesn't need to be solved now as the current one doesn't need it though)

Reply:
I don't think the block directory handles that case, right now. That doesn't have to exclude those plugins from here, though--single block plugins are just the first focus. I'll tweak the language.